File: src/main/java/no/novari/instance/gateway/model/acos/AcosInstance.java
package no.novari.instance.gateway.model.acos;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.extern.jackson.Jacksonized;
import no.novari.flyt.instance.gateway.validation.constraints.ValidBase64;
import no.novari.instance.gateway.validation.UniqueElementIds;

import java.util.ArrayList;
import java.util.List;

@Getter
@EqualsAndHashCode
@Jacksonized
@Builder
public class AcosInstance {

    @NotNull
    @Valid
    private AcosInstanceMetadata metadata;

    @NotEmpty
    @UniqueElementIds
    @Valid
    private List<@NotNull AcosInstanceElement> elements;

    @NotBlank
    @ValidBase64
    private String formPdfBase64;

    @Builder.Default
    @Valid
    private List<@NotNull AcosDocument> documents = new ArrayList<>();
}

File: src/main/kotlin/no/novari/flyt/acos/instance/gateway/model/acos/AcosInstance.kt
package no.novari.flyt.acos.instance.gateway.model.acos

import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.NotNull
import no.novari.flyt.gateway.instance.validation.constraints.ValidBase64

data class AcosInstance(
    @field:NotNull
    @field:Valid
    val metadata: AcosInstanceMetadata,
    @field:NotEmpty
    @field:Valid
    val elements: List<@NotNull AcosInstanceElement>,
    @field:NotBlank
    @field:ValidBase64
    val formPdfBase64: String,
    @field:Valid
    val documents: List<@NotNull AcosDocument> = emptyList(),
)
